### PR TITLE
Fix issue #163: briefer why comments

### PR DESCRIPTION
**Pull request by kimgr**
Oops, I seem to have killed the old PR. Here's an updated one with review comments addressed.

### CODE REVIEW

File: iwyu_output.cc [hunk 1]
```diff
@@ -72,53 +72,60 @@ class OutputLine {
...
-      ++symbols_printed;
-      prefix = ", ";       // before 2nd and subsequent symbols, print ', '
-    } else {
-      // Truncate at max_length cols.
-      if (symbols_printed > 0)
-        retval += ", etc";
-      break;
+// Append a helpful 'why' comment to the include line, containing the symbols
+// used from the header. Align nicely at lower verbosity levels.
+string OutputLine::printable_line(size_t min_length, size_t max_length) const {
```
Comment (vsapsai) [inline iwyu_output.cc]: Should we have some CHECK_ that max_length >= min_length and that max_length > 0 or some other reasonable value?

File: iwyu_output.cc [hunk 2]
```diff
@@ -72,53 +72,60 @@ class OutputLine {
...
+  if (symbols_.empty())
+    return line_;
+
+  --max_length;  // Spare a char for newline character.
+
+  // Before first symbol, print '  // for ' and pad it so 'why' comments are
+  // nicely aligned.
+  string symbol_prefix = "  // for ";
+  if (min_length > line_.length())
+    symbol_prefix.insert(0, min_length - line_.length(), ' ');
```
Comment (vsapsai) [inline iwyu_output.cc]: I find `if (line_.length() < min_length)` more readable. Cannot explain why, maybe it is just a personal taste. It is up to you to decide in which order to write this comparison.

File: iwyu_output.cc [hunk 3]
```diff
@@ -72,53 +72,60 @@ class OutputLine {
...
+  string result = line_;
+  for (string symbol : symbols_) {
+    // At verbose levels 0-2, truncate output to max_length columns.
+    if (!ShouldPrint(3)) {
+      // Calculate number of chars remaining.
+      size_t remaining = 0;
+      size_t result_length = result.length() + symbol_prefix.length();
+      if (max_length > result_length)
+        remaining = max_length - result_length;
+
```
Comment (vsapsai) [inline iwyu_output.cc]: And here too, I prefer `result_length < max_length`. Interesting, if there is statistic data regarding usage of `<` and `>`.

**Comment by vsapsai** [comment]
Code looks correct, just a few suggestions.

File: iwyu_output.cc (hunk 1 above)
Comment (kimgr) [inline iwyu_output.cc]: `max_length > 0` is required for correctness, so I added a `CHECK_` for that.

I think it's legal for `min_length` to be larger than `max_length`, however, e.g. if an `#include` line exceeds the configured max length, and that should be handled by the algorithm as-is.

**Comment by kimgr** [comment]
Thanks, updated the diff with the conditions turned around and a `CHECK_` added.

File: iwyu_output.cc (hunk 1 above)
Comment (vsapsai) [inline iwyu_output.cc]: Good point that `min_length` can be larger than `max_length`.

**Comment by vsapsai** [comment]
No more comments, you are good to go.

**Pull request merged**